packages/fx-core/src/plugins/resource/apim/managers/apimManager.ts

```diff
@@ -207,53 +215,65 @@ export class ApimManager {
     await apimService.addApiToProduct(resourceGroupName, apimServiceName, productId, apiId);
   }
 
-  public async updateArmTemplates(): Promise<ArmTemplateResult> {
+  public async updateArmTemplates(ctx: PluginContext): Promise<ArmTemplateResult> {
+    const azureSolutionSettings = ctx.projectSettings?.solutionSettings as AzureSolutionSettings;
```

Comment:
what will happen if projectSettings is null or undefined? Any check  needed for azureSolutionSettings for followingsteps?

Reply:
this interface only called from solution, solution will parse ctx first.(https://github.com/OfficeDev/TeamsFx/blob/b985a35981afd09f3aafe84462f26acd42342685/packages/fx-core/src/plugins/solution/fx-solution/arm.ts?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L768)
and plugins invoke these interface with the same ctx.